### ISSUE TITLE
New rule `vue/padded-blocks`

### ISSUE THREAD

**Issue by sebj54** [vuejs/eslint-plugin-vue#2199]
**Please describe what the rule should do:**

The rule should enforce no padding in top-level blocks (template, script & style). This rule is like this one: https://eslint.org/docs/latest/rules/padded-blocks, except it applies only on top-level tags

Example with 

**What category should the rule belong to?**

[x] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

`"vue/padded-blocks": ["error", "never"]`

```vue
<!-- BAD -->
<script>

a()
</script>

<script>
a()

</script>

<script>

a()

</script>

<template>

  <div />
</template>

<template>
  <div />

</template>

<template>

  <div />

</template>

<style>

.styled {}
</style>

<style>
.styled {}

</style>

<style>

.styled {}

</style>

<!-- GOOD -->
<script>
a()
</script>

<template>
  <div />
</template>

<style>
.styled {}
</style>
```

I think the "always" option is relevant too, but I think it would need to be configurable, for instance by specifying which top-level blocks should have padding and if padding should be on first and/or last line.

**Comment by ota-meshi** [comment]
Is `vue/block-tag-newline` rule what you're looking for?

https://eslint.vuejs.org/rules/block-tag-newline.html

**Comment by sebj54** [comment]
Oops, I missed it before posting the issue! You're right, it's what I needed, especially with the `maxEmptyLines` option. 

Thank you very much!
